js_modules/dagster-ui/packages/ui-core/src/workers/WorkerThread.oss.ts

```diff
@@ -0,0 +1,23 @@
+import {WEB_WORKER_FEATURE_FLAGS_KEY, setFeatureFlags} from '../app/Flags';
+
+export const createWorkerThread = (
+  onMessage: (postMessage: (message: any) => void, data: any) => Promise<void>,
+  onError: (postMessage: (message: any) => void, error: Error) => void,
+) => {
```

Comment:
I ended up making `onMessage`/`onError` take `postMessage` as an argument because I want to avoid the workers calling `self.postMessage` directly. The thinking is that by avoiding using `self` directly I should be able to create a mock version `createWorkerThread` that uses the real worker code directly creating a simpler mocking experience. I think to do this we might need to generate code though, at the very least when special behavior in jest...  I'm still thinking through that part but wanted to put this PR up since I have an immediate use case